tests/mlmodel_openai/test_chat_completion.py

```diff
@@ -13,19 +13,184 @@
+            'response.headers.ratelimitResetTokens': "90ms",
+            'response.headers.ratelimitResetRequests': "7m12s",
+            'response.headers.ratelimitRemainingTokens': 39940,
+            'response.headers.ratelimitRemainingRequests': 199,
+            'vendor': 'openAI',
+            'ingest_source': 'Python',
+            'number_of_messages': 3,
+        },
+    ),
+    (
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-0",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': 'You are a scientist.',
+            'role': 'system',
+            'completion_id': None,
+            'sequence': 0,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        },
+    ),
+    (
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-1",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': 'What is 212 degrees Fahrenheit converted to Celsius?',
+            'role': 'user',
+            'completion_id': None,
+            'sequence': 1,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        },
+    ),
+    (
+        {'type': 'LlmChatCompletionMessage'},
+        {
+            'id': "chatcmpl-87sb95K4EF2nuJRcTs43Tm9ntTemv-2",
+            'appName': 'Python Agent Test (mlmodel_openai)',
+            'request_id': "49dbbffbd3c3f4612aa48def69059ccd",
+            'span_id': "span-id",
+            'trace_id': "trace-id",
+            'transaction_id': None,
+            'content': '212 degrees Fahrenheit is equal to 100 degrees Celsius.',
+            'role': 'assistant',
+            'completion_id': None,
+            'sequence': 2,
+            'response.model': 'gpt-3.5-turbo-0613',
+            'vendor': 'openAI',
+            'ingest_source': 'Python'
+        }
+    ),
+]
+
+
+@reset_core_stats_engine()
+@validate_ml_events(sync_chat_completion_recorded_events)
+# One summary event, one system message, one user message, and one response message from the assistant
+@validate_ml_event_count(count=4)
+@background_task()
+def test_openai_chat_completion_sync_in_txn():
+    set_trace_info()
+    add_custom_attribute("conversation_id", "my-awesome-id")
+    openai.ChatCompletion.create(
+        model="gpt-3.5-turbo",
+        messages=_test_openai_chat_completion_sync_messages,
+        temperature=0.7,
+        max_tokens=100
+    )
+
+
```

Comment:
We should have a test in here for when conversation_id isn't set on the transaction.